### PR TITLE
add support for job object management

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: winapi/jobobj.go [hunk 1]
```diff
@@ -35,13 +35,54 @@ const (
...
 )
 
 type JOBOBJECT_ASSOCIATE_COMPLETION_PORT struct {
 	CompletionKey  uintptr
 	CompletionPort syscall.Handle
 }
 
+type IO_COUNTERS struct {
+	ReadOperationCount  uint64
+	ReadOperationCount  uint64
```
Comment (alexbrainman) [inline winapi/jobobj.go]: Why duplicate ReadOperationCount field?

File: winapi/jobobj.go [hunk 2]
```diff
@@ -35,13 +35,54 @@ const (
...
+	WriteOperationCount uint64
+	OtherOperationCount uint64
+	ReadTransferCount   uint64
+	WriteTransferCount  uint64
+	OtherTransferCount  uint64
+}
+
+type JOBOBJECT_BASIC_LIMIT_INFORMATION struct {
+	PerProcessUserTimeLimit uint64
```
Comment (alexbrainman) [inline winapi/jobobj.go]: s/uint64/int64/

No?

PerProcessUserTimeLimit is a LARGE_INTEGER. But LARGE_INTEGER is "A 64-bit signed integer. The range is –9223372036854775808 through 9223372036854775807 decimal." according to https://msdn.microsoft.com/en-us/library/windows/desktop/aa383751(v=vs.85).aspx

File: winapi/jobobj.go [hunk 3]
```diff
@@ -35,13 +35,54 @@ const (
...
+	PerJobUserTimeLimit     uint64
```
Comment (alexbrainman) [inline winapi/jobobj.go]: Same.

**Review by alexbrainman — COMMENTED** [review]
Sounds reasonable.

I would like to know why add these structures here? How do you use them?

Thank you

Alex

**Comment by tychoish** [comment]
You can pass them to `SetInformationJobObject` to control the behavior of job objects.

I resolved the issues you raised in the code review. 

Sorry for being terse earlier. I'm working on a project that needs to set some options on a Job Object, which this project provides nearly *all* of the required options, but not quite (e.g. what I added here.) So I wanted to put them out there for the next person in my position.

**Comment by alexbrainman** [comment]
> I'm working on a project that needs to set some options on a Job Object, which this project provides nearly all of the required options, but not quite (e.g. what I added here.)

Thank you for explaining.

Alex

**Pull request merged**